src/transformers/agents/agents.py

```diff
@@ -488,26 +510,35 @@ def execute_tool_call(self, tool_name: str, arguments: Dict[str, str]) -> Any:
             tool_name (`str`): Name of the Tool to execute (should be one from self.toolbox).
             arguments (Dict[str, str]): Arguments passed to the Tool.
         """
-        if tool_name not in self.toolbox.tools:
-            error_msg = f"Error: unknown tool {tool_name}, should be instead one of {list(self.toolbox.tools.keys())}."
+        available_tools = self.toolbox.tools
+        if self.managed_agents is not None:
+            available_tools = {**available_tools, **self.managed_agents}
```

Comment:
looks like this part can be moved to self.toolbox() since it's a dedicated function for that

Reply:
I'd prefer to leave managed agents as a separate thing from Tools, since they're supposed to be handled differently here.